internal/controllers/rest_controller.go

```diff
@@ -7,49 +7,102 @@ import (
 
 	"github.com/gin-gonic/gin"
 
+	"github.com/ydataai/authentication-service/internal/configurations"
+	"github.com/ydataai/authentication-service/internal/handlers"
+	"github.com/ydataai/authentication-service/internal/models"
 	"github.com/ydataai/authentication-service/internal/services"
 	"github.com/ydataai/go-core/pkg/common/logging"
 	"github.com/ydataai/go-core/pkg/common/server"
 )
 
 // RESTController defines rest controller.
 type RESTController struct {
-	configuration RESTControllerConfiguration
-	oidcService   *services.OIDCService
+	configuration configurations.RESTControllerConfiguration
+	oidcService   services.OIDCService
+	credentials   []handlers.CredentialsHandler
 	logger        logging.Logger
 }
 
 // NewRESTController initializes rest controller.
 func NewRESTController(
 	logger logging.Logger,
-	configuration RESTControllerConfiguration,
-	oidcService *services.OIDCService,
+	configuration configurations.RESTControllerConfiguration,
+	oidcService services.OIDCService,
+	credentials []handlers.CredentialsHandler,
 ) RESTController {
 	return RESTController{
 		configuration: configuration,
 		oidcService:   oidcService,
+		credentials:   credentials,
 		logger:        logger,
 	}
 }
 
 // Boot initialize creating some routes.
 func (rc RESTController) Boot(s *server.Server) {
-	s.Router.GET(rc.configuration.AuthServiceURL, gin.WrapF(rc.RedirectToOIDCProvider))
+	s.Router.GET(rc.configuration.AuthServiceURL, gin.WrapF(rc.AuthenticationSession))
 	s.Router.GET(rc.configuration.OIDCCallbackURL, gin.WrapF(rc.OIDCProviderCallback))
+	s.Router.GET(rc.configuration.LogoutURL, gin.WrapF(rc.Logout))
+}
+
+// AuthenticationSession is responsible for knowing if the user already has a valid credential or not.
+// If so, forward 200 OK + UserID Headers.
+// If not, begin OIDC Flow.
+func (rc RESTController) AuthenticationSession(w http.ResponseWriter, r *http.Request) {
+	// workflow to identify if there is a token present.
+	token, err := rc.getCredentials(r)
+	// if a token is not identified, the OIDC flow will be started.
+	if err == nil && token == "" {
+		rc.RedirectToOIDCProvider(w, r)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	claims, err := rc.oidcService.Decode(token)
```

Comment:
from this decode, you only retrieve the claims? 🤔

Reply:
yeap! do you've another idea for this?

Reply:
we validate the token,
if so, we return the claims.

Reply:
We can named it as TokenInfo or UserInfo.

Reply:
I won't need the other property anymore `claims["access_token"] = tokenString`.
because I already have the token in the upstream.